Mathlib/Topology/PartialHomeomorph.lean

```diff
@@ -1346,6 +1346,15 @@ noncomputable def toPartialHomeomorph [Nonempty α] : PartialHomeomorph α β :=
     h.continuous.continuousOn h.isOpenMap isOpen_univ
 #align open_embedding.to_local_homeomorph OpenEmbedding.toPartialHomeomorph
 
+/-- An embedding defines a homeomorphism to its range. -/
+noncomputable def _root_.Embedding.homeomorph : Homeomorph α (range f) := by
+  by_cases hX : Nonempty α
+  · exact (Homeomorph.Set.univ α).symm.trans <|
+      (h.toPartialHomeomorph f).homeomorphOfImageSubsetSource Subset.rfl image_univ
+  · rw [not_nonempty_iff] at hX
+    have : IsEmpty (range f) := by rw [isEmpty_coe_sort, range_eq_empty f]
+    exact Homeomorph.empty
```

Comment:
This def isn't definitionally nice. I think the Equiv part should be given by [Equiv.ofInjective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Equiv/Set.html#Equiv.ofInjective), but you're free to split by cases etc. to prove continuity.

Reply:
We have [Homeomorph.ofEmbedding](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Homeomorph.html#Homeomorph.ofEmbedding)

Reply:
Nice find. I've dropped that lemma.